Api/RequestHandler.php

```diff
@@ -196,12 +200,16 @@ protected function decryptLast4( $payment )
         if (strlen($last4) == 4 && ctype_digit($last4)) {
             return $last4;
         }
+
+        if ( strlen($last4) > 4 || empty($last4) ){
```

Comment:
Could be worth changing this to an `else` or `elseif` I have the filling we aren't being exhaustive with the options here and this could lead to unexpected behaviour, also this would make the return block clearer it is either one or the other.